### PR TITLE
fix: show cache indication whenever cache is available

### PR DESCRIPTION
**Pull request by Hill-98**
#239

### CODE REVIEW

File: scripts/uosc.lua [hunk 1]
```diff
@@ -4172,15 +4172,18 @@ end))
 mp.observe_property('demuxer-cache-state', 'native', function(prop, cache_state)
 	local cached_ranges = cache_state and cache_state['seekable-ranges'] or {}
 	local uncached_ranges = nil
-	if state.duration and state.is_stream then
+	if state.duration then
```
Comment (christoph-heinrich) [inline scripts/uosc.lua:4175]: Wouldn't it be simpler to just check for `#cached_ranges > 0` here instead of counting in the for?

**Comment by christoph-heinrich** [comment]
This brings up the question how `state.is_stream` should be handled. arguably it should always be true when there is a cache or not?

File: scripts/uosc.lua (hunk 1 above)
Comment (Hill-98) [reply scripts/uosc.lua:4175]: I'm not a Lua programmer, I found this method when I searched the web how to get the table size, I'll change it when I wake up.

**Comment by Hill-98** [comment]
> This brings up the question how `state.is_stream` should be handled. Arguably it should always be true when there is a cache.

I'm used to setting `--cache` to `yes` so that it will be cached no matter what, but it doesn't have to be a stream.

**Comment by tomasklaen** [comment]
Add:

```lua
if not state.is_stream then
	set_state('is_stream', true)
	Elements:trigger('dispositions')
end
```

at the end of the `if state.duration and #cached_ranges > 0 then` condition.

**Comment by Hill-98** [comment]
As I said above, there is not necessarily a 'stream' when there is a cache.

So I feel like either get rid of `is_stream`, or keep the status quo, and there are not particularly many features currently relying on `is_stream`.

**Comment by christoph-heinrich** [comment]
This now overwrites the full cached range set by the `demuxer-via-network` observer for early display of uncached range ._.
That's why I suggested adding the `#cached_ranges > 0` check before the other if :neutral_face:

**Comment by tomasklaen** [comment]
Oh I misunderstood your request. I saw `#cached_ranges > 0` in the updated PR and thought that was it and merged 🤷‍♂️. I'll fix it.

**Pull request merged**